api.v2/java.v2/src/main/java/com/listeners/TestNGListenerForPractiTest.java

```diff
@@ -0,0 +1,48 @@
+package com.listeners;
+
+import com.practitest.api.example.PractiTestWriter;
+import com.practitest.integration.ExtractTests;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+
+import java.util.List;
+
+public class TestNGListenerForPractiTest implements ITestListener {
+
+
+        @Override
+        public void onTestStart(ITestResult result) {
+        }
+
+        @Override
+        public void onTestSuccess(ITestResult result) {
+        }
+
+        @Override
+        public void onTestFailure(ITestResult result) {
+        }
+
+        @Override
+        public void onTestSkipped(ITestResult result) {
+        }
+
+        @Override
+        public void onTestFailedButWithinSuccessPercentage(ITestResult result) {
+        }
+
+        @Override
+        public void onStart(ITestContext context) {
+                List<Integer> testIDs = ExtractTests.extractAllTestIds(context);
+                //Create test run for all tests in current execution
+                Integer setID = PractiTestWriter.createNewSet(testIDs);
+                //Store SetID for further usage
+                System.setProperty("currentSetId", setID.toString());
+                PractiTestWriter.createNewSet(testIDs);
```

Comment:
Not sure i understand the logic here. Are you going to create a new TestSet every time 'mvn test' runs?

Reply:
As I understand TestSet this is collection of tests which should be executed.
For example if you specify certain test for execution you'll need new TestSet in Practitest to store results for it. Correct me if I'm wrong

Reply:
You right, TestSet is a collection of Tests. But subsequent executions are supposed to reuse existing TestSet. 
Meaning that when you run your project first time, we should create the Tests and TestSet. When you run the project next time, we should use the TestSet created during first run.

Reply:
Most of the time automated and manual tests are grouped to Smoke and Regression test suites or by some other project specific definition.
This means we should have multiple TestSets for them. Including test automation process into consideration we will have only some part automated at the start and number of automated != manual test cases. From this perspective it seam to me logical to create new TestSet each time as collection of tests most of the time will be different.

Reply:
I can add validation to reuse test set if collection of tests already exist for this group. Please let me know if this will work for you.

Reply:
Yes, I think we should reuse testsets. If tests are added or removed, we should change the existing testset.